jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/SymlinkAllowedResourceAliasChecker.java

```diff
@@ -57,29 +57,33 @@ protected boolean check(String pathInContext, Path path)
         // We rebuild the realURI, segment by segment, getting the real name at each step, so that we can distinguish between
         // alias types.  Specifically, so we can allow a symbolic link so long as it's realpath is not protected.
         String[] segments = pathInContext.substring(1).split("/");
-        Path fromBase = _base;
-        StringBuilder realURI = new StringBuilder();
+        StringBuilder segmentPath = new StringBuilder();
 
         try
         {
             for (String segment : segments)
             {
                 // Add the segment to the path and realURI.
-                fromBase = fromBase.resolve(segment);
-                realURI.append("/").append(fromBase.toRealPath(NO_FOLLOW_LINKS).getFileName());
+                segmentPath.append("/").append(segment);
+                Resource fromBase = _baseResource.resolve(segmentPath.toString());
```

Comment:
This line has a different behaviour than before which is causing the change to the test at `DefaultServletTest#testGetPercent2F`. Before we were resolving a path, this caused the `toRealPath` would throw because it still had the `%2F` which did not exist.

However the `Resource.resolve()` resolves as a URI then does a `Paths.get()` which resolves the `%2F` to a `/`.